pkg/transparentproxy/config/config_executables.go

```diff
@@ -148,11 +133,23 @@ type InitializedExecutable struct {
 	args []string
 }
 
-func (c InitializedExecutable) Exec(
-	ctx context.Context,
-	args ...string,
-) (*bytes.Buffer, *bytes.Buffer, error) {
-	return execCmd(ctx, c.logger, c.cniMode, c.NeedLock(), c.Path, append(c.args, args...)...)
+func (c InitializedExecutable) NeedLock() bool {
+	// iptables-nft does not use the xtables lock, so no lock is needed for this
+	// mode
+	if c.version.Mode == IptablesModeNft {
+		return false
+	}
+
+	// Only iptables and iptables-restore executables need a lock because they
+	// perform write operations
+	switch c.name {
+	case "", "restore":
+		return true
+	}
+
+	// For all other cases (such as iptables-save), no lock is needed as they
+	// don't perform write operations
+	return false
```

Comment:
This method is exactly the same as before, but was just moved from `Executable` to `InitializedExecutable`